### DISCUSSION TITLE
Cannot get RDP Connect or Web-RDP to work against gnome-remote-desktop on Ubuntu Weyland - RDP on PC does work

### DISCUSSION THREAD

**Discussion by davwoo** [Ylianst/MeshCentral#6437] [Issues] [answered]
Firstly, humble apologies if this has been addressed elsewhere. Slap me down passive-aggressively with links if you wish, but I have been looking in here and in the Google-verse for the last couple of days and have found various references to RDP/Gnome/Weyland/Web-RDP/black-screens, but I can't find anything that has fixed this particular flavour of the issue. So, without further ado...

_**My setup:**_

**_MeshCentral:_**

- v1.1.32, happily running in LANonly mode on a Raspberry Pi Model B Plus Rev 1.2, under Raspbian GNU/Linux 11 (bullseye).

_**Target device.** For the sake of brevity, let's call it, _FfsLetMeIn_:_
- A repurposed HP Pavilion X360 (Core i5, 7th Gen.) running Ubuntu 24.04.1 with the default Gnome (Weyland) desktop (v46), on the same LAN as the MeshCentral server
- Using using FfsLetMeIn's menus I have enabled (gnome-remote-desktop) _Settings > System > 'Desktop sharing'_ and _'Remote control'_ on port 3389, but not _'Remote login'_ - I am only interested in controlling my current session, not in starting a new one. That said, I did try Remote control with the same results as below
- I can successfully control the current session on FfsLetMeIn using the default Remote Desktop Connection app from a Windows 11 PC on the same LAN, so I'm fairly happy that RDP access is possible

In MeshCentral I have two devices for FfsLetMeIn, one is a "Devices with no-agent" and the other is a Linux MeshCentral Linux agent.

_**The Linux agent:**_

- Allows me to connect via Terminal
- Attempted connection via the default Desktop gives the expected "Bleah! Wayland! Use Xorg!" type message. Sorry, no can do
- Attempted connection via Web/RDP gives a login page that, when I click on Connect, goes black for _four or five seconds_ then returns to the login page

_**The Devices with no-agent... not an-agent:**_

- Attempted connection via the Desktop RDP Connect button (with the correct credentials) gives a black screen and "setup..." appears next to the button. After about 30 seconds, it goes back to "Disconnected"
- Attempted connection via Web/RDP gives the login page that, when I click on Connect, goes black this time for about _30 seconds_ then returns to the login page.

I'm currently at a loss, so any pointers, assistance or general prodding in the right direction would be greatly appreciated.

**Comment by si458** [comment, marked as answer]
this is a duplicate of issues already open
https://github.com/Ylianst/MeshCentral/issues/6384
https://github.com/Ylianst/MeshCentral/issues/4927
https://github.com/Ylianst/MeshCentral/issues/3950
sadly the Web-RDP doesnt seem to work with Linux anymore while it use to,
so im guessing something changged in the rdp server side that the web-rdp no longer likes,
so we need to upgrade.fix the web-rdp but its not simple because the rdp is a microsoft protocol
so we need a microsoft export to look into why it no longer works

**Reply by davwoo** [reply]
Thanks for letting me know and for the links. 🙂 I saw #3950 but my brain didn’t make the connection. Sorry.

I was reading that [gnome-remote-desktop used to be compiled with VNC support built in](https://askubuntu.com/questions/1469028/gnome-screen-sharing-vnc-23-04), but it was decided to remove it. If I were to compile from source with VNC enabled (apparently something that can be done) do you think Web-VNC might stand a chance of working?

Or does anyone know of a Weyland compatible VNC server?

**Reply by si458** [reply]
Any vnc server installed on ur Linux machine will work!
I use the built in realvnc that comes with raspberry pi
Also also use i think it was tigervnc with my Ubuntu 20.04 vm!
So long as u can vnc u can use the web-vnc option :)
Also note, if u use wayland, u can switch back to X11, which will then restore the desktop tab and allow control of the machine without rdp or vnc

**Reply by davwoo** [reply]
Thanks. 🙂

X11 is, unfortunately, not an option in this instance. ☹️

**Reply by si458** [reply]
@davwoo any reason why u can't switch back to x11? U can run wayland and x11 side by side, even switch between them when u login (well ubuntu u can anyways)

**Reply by davwoo** [reply]
@si458 : Policy set by those further up the food chain, though having just learned that you can run Way & X side by side could be a possible loophole. 😉 I shall investigate further. Thanks for the suggestions. 🙂

**Reply by si458** [reply]
@davwoo ah yes the higher ups! Always a pain in the ass haha!

from my testing on Ubuntu 22.04 previously, I could set it to use wayland, then u can setup X11, then when u login u get an option to pick weather x11 or wayland

but also u can force Ubuntu to use x11 always but still have the ability to logout and use wayland if u wanted!

It will be a case of trial and error and report back!

**Reply by davwoo** [reply]
@si458 : I’ve got it on my todo list. 🙂

**Reply by davwoo** [reply]
@si458 After spending a couple of fruitless days trying to find a workaround and a version of VNC server that will reliably install under Wayland, it was deemed that I was "spending too much time on this" and hence was grated an official temporary stay on the policy. I am now _temporarily_ ;) running X.Org instead of Wayland "until the RDP issue with Wayland is resolved" so, result but no result. ¯\\_(ツ)_/¯.

Thx for the assistance. I'll keep an eye on the duplicate issues. Which one is the lead on this subject, would you say?
